gclient/src/api/calls.rs

```diff
@@ -222,20 +347,82 @@ impl GearApi {
         for event in events.iter() {
             if let Event::Gear(GearEvent::MessageEnqueued {
                 id,
-                source,
-                entry: Entry::Reply(replied_on),
+                entry: Entry::Reply(_),
                 ..
             }) = event?.event
             {
-                if &source == expected_source && MessageId::from(replied_on) == reply_to_id {
-                    return Ok((id.into(), message.value(), tx.block_hash()));
-                }
+                return Ok((id.into(), message.value(), tx.block_hash()));
             }
         }
 
         Err(Error::EventNotFound)
     }
 
+    /// `pallet_gear::send_reply` with bytes in payload.
+    pub async fn send_reply_bytes_batch(
+        &self,
+        args: impl IntoIterator<Item = (MessageId, impl AsRef<[u8]>, u64, u128)>,
+    ) -> Result<(Vec<Result<(MessageId, u128)>>, H256)> {
+        let mut reply_to_ids = vec![];
```

Comment:
I guess you can avoid this additional allocation too by using the `args` iterator.